control-plane/plugin/src/resources/snapshot.rs

```diff
@@ -117,3 +125,85 @@ async fn get_snapshots(
 
     Ok(snapshots)
 }
+
+#[async_trait(?Send)]
+impl GetSnapshotTopology for VolumeSnapshots {
+    type SourceID = Option<VolumeId>;
+    type ResourceID = Option<SnapshotId>;
+    async fn get_snapshot_topology(
+        volid: &Self::SourceID,
+        snapid: &Self::ResourceID,
+        output: &utils::OutputFormat,
+    ) -> PluginResult {
+        match get_snapshots(volid, snapid).await {
+            Ok(snapshots) => {
+                let snapshot_topology = VolumeSnapshotTopology(snapshots);
+                // Print table, json or yaml based on output format.
+                utils::print_table(output, snapshot_topology);
+            }
+            Err(e) => return Err(e),
+        }
+        Ok(())
+    }
+}
+impl GetHeaderRow for VolumeSnapshotTopology {
+    fn get_header_row(&self) -> Row {
+        utils::SNAPSHOT_TOPOLOGY_PREFIX
+            .iter()
+            .chain(utils::SNAPSHOT_TOPOLOGY_HEADERS.iter())
+            .cloned()
+            .collect()
+    }
+}
+
+impl CreateRows for VolumeSnapshotTopology {
+    fn create_rows(&self) -> Vec<Row> {
+        self.0
+            .iter()
+            .flat_map(|volume| {
+                let mut rows = Vec::new();
+                volume
+                    .state
+                    .replica_snapshots
+                    .create_rows()
+                    .into_iter()
+                    .enumerate()
+                    .for_each(|(i, mut r)| {
+                        let mut row = if i == 0 {
+                            row![volume.state.uuid]
+                        } else if i < volume.state.replica_snapshots.len() - 1 {
+                            row!["├─"]
+                        } else {
+                            row!["└─"]
+                        };
+                        for cell in r.iter_mut() {
+                            row.add_cell(cell.clone());
+                        }
+                        rows.push(row);
+                    });
+                rows
+            })
+            .collect()
+    }
+}
+impl CreateRows for Vec<openapi::models::ReplicaSnapshotState> {
+    fn create_rows(&self) -> Vec<Row> {
+        self.iter()
+            .map(|state| match state {
+                openapi::models::ReplicaSnapshotState::online(s) => {
+                    row![
+                        s.uuid,
+                        s.pool_id,
+                        "Online",
+                        s.size,
+                        s.allocated_size,
+                        s.source_id
+                    ]
+                }
+                openapi::models::ReplicaSnapshotState::offline(s) => {
+                    row![s.uuid, s.pool_id, "Offline", "", "", s.source_id]
+                }
```

Comment:
Not sure if this will be allowed, but you can give it a try something like below to do in single statement without `match`
```
row![s.uuid, s.pool_id, format!("{}", if openapi::models::ReplicaSnapshotState::online "Online" else "Offline"), "", "", s.source_id]
```

Reply:
For Offline, We will show the size and allocated_size field. I feel keeping it separate will have a better readability.